assets/src/scripts/monitoring-location/iv-data-utils.js

```diff
@@ -132,29 +142,38 @@ const createConvertedVariable = function(NWISVariable) {
 *   @prop - {Object) points - An object containing temperature data
 *   @prop - {String} variable - The 'variable' from the NWIS system, will have a suffix indicating a converted value
  */
-const createConvertedTimeSeries = function(timeSeries, tsRequestKey, parameterCode) {
-    timeSeries.points.forEach(temperaturePoint => {
-        temperaturePoint.value = convertCelsiusToFahrenheit(temperaturePoint.value).toFixed(2);
-    });
-
-    timeSeries.variable = `${timeSeries.variable}${config.CALCULATED_TEMPERATURE_VARIABLE_CODE}`;
-    const tsRequestKeyValue = `${tsRequestKey}:${parameterCode}${config.CALCULATED_TEMPERATURE_VARIABLE_CODE}`;
-
+const getConvertedTimeSeries = function(timeSeries, tsRequestKey, parameterCode) {
+    const fahrenheitTimeSeries = {
+        ...timeSeries,
+        censorCode: [...timeSeries.censorCode],
+        offset: [...timeSeries.offset],
+        qualifier: [...timeSeries.qualifier],
+        qualityControlLevel: [...timeSeries.qualityControlLevel],
+        sample: [...timeSeries.sample],
+        source: [...timeSeries.source],
+        points: timeSeries.points.map((point) => {
+            return {
+                ...point,
+                qualifiers: [...point.qualifiers],
+                value: point.value ? convertCelsiusToFahrenheit(point.value).toFixed(2) : null
```

Comment:
This is how I handled setting Fahrenheit to null when the celsius value is null.

Reply:
oh nice. That is a couple of lines less code than what I had done.